src/services_models/reserves.jl

```diff
@@ -143,3 +145,35 @@ function _device_services!(
     end
     return
 end
+
+function add_device_services!(
+    constraint_data_in::RangeConstraintsData,
+    constraint_data_out::RangeConstraintsData,
+    device::D,
+    model::DeviceModel{D, <:AbstractStorageFormulation},
+) where {D <: PSY.Storage}
+    for service_model in get_services(model)
+        if PSY.has_service(device, service_model.service_type)
+            services =
+                [s for s in PSY.get_services(device) if isa(s, service_model.service_type)]
```

Comment:
You don't actually need a vector. You could use a generator instead.

Minor readability thing:  I find `s isa service_model.service_type` more readable.  Just my opinion.

Reply:
we use isa() as a function throughout the code in PSI.